fluss-server/src/main/java/com/alibaba/fluss/server/metadata/CoordinatorServerMetadataCache.java

```diff
@@ -0,0 +1,87 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.fluss.server.metadata;
+
+import com.alibaba.fluss.cluster.ServerNode;
+import com.alibaba.fluss.metadata.TablePath;
+import com.alibaba.fluss.server.coordinator.CoordinatorContext;
+import com.alibaba.fluss.server.coordinator.CoordinatorServer;
+
+import javax.annotation.Nullable;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/** The implement of {@link ServerMetadataCache} for {@link CoordinatorServer}. */
+public class CoordinatorServerMetadataCache implements ServerMetadataCache {
+
+    private final CoordinatorContext coordinatorContext;
+
+    public CoordinatorServerMetadataCache(CoordinatorContext coordinatorContext) {
+        this.coordinatorContext = coordinatorContext;
+    }
+
+    @Override
+    public boolean isAliveTabletServer(int serverId) {
+        Map<Integer, ServerInfo> aliveTabletServer = coordinatorContext.getLiveTabletServers();
+        return aliveTabletServer.containsKey(serverId);
+    }
+
+    @Override
+    public Optional<ServerNode> getTabletServer(int serverId, String listenerName) {
+        Map<Integer, ServerInfo> aliveTabletServer = coordinatorContext.getLiveTabletServers();
+        return aliveTabletServer.containsKey(serverId)
+                ? Optional.ofNullable(aliveTabletServer.get(serverId).node(listenerName))
+                : Optional.empty();
+    }
+
+    @Override
+    public Map<Integer, ServerNode> getAllAliveTabletServers(String listenerName) {
+        Map<Integer, ServerNode> serverNodes = new HashMap<>();
+        for (Map.Entry<Integer, ServerInfo> entry :
+                coordinatorContext.getLiveTabletServers().entrySet()) {
+            ServerNode serverNode = entry.getValue().node(listenerName);
+            if (serverNode != null) {
+                serverNodes.put(entry.getKey(), serverNode);
+            }
+        }
+        return serverNodes;
+    }
+
+    @Override
+    public @Nullable ServerNode getCoordinatorServer(String listenerName) {
+        ServerInfo coordinatorServer = coordinatorContext.getCoordinatorServerInfo();
+        return coordinatorServer == null ? null : coordinatorServer.node(listenerName);
+    }
+
+    @Override
+    public Set<Integer> getAliveTabletServerIds() {
+        return Collections.unmodifiableSet(coordinatorContext.getLiveTabletServers().keySet());
+    }
+
+    @Override
+    public Optional<TablePath> getTablePath(long tableId) {
+        return Optional.ofNullable(coordinatorContext.getTablePathById(tableId));
```

Comment:
Don't call method `coordinatorContext.getTablePathById(tableId)` in here since it'll cause `coordinatorContext` accessed in mutiple threads, but `coordinatorContext` is not thread safe.

Reply:
Client's get `metadata` request can also reach the coordinator server, then the coordinator server will call `CoordinatorServerMetadataCache#getTablePath ` to get the table path from `coordinatorContext`. But there's another separate thread in `CoordinatorEventProcessor` that will also update the table path from `coordinatorContext`. But the `coordinatorContext` is thread unsafe.